### PR TITLE
SD-562: write integration test for payout

### PR DESCRIPTION
**Pull request by domagojlatecki**

### CODE REVIEW

File: contracts/services/PayoutManager.sol [hunk 1]
```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./IMerkleTreePathValidator.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+contract PayoutManager {
```
Comment (githinho) [inline contracts/services/PayoutManager.sol:8]: Don't forget to update after changes requested by Filip.

File: test/full-test.ts [hunk 2]
```diff
@@ -379,6 +381,158 @@ describe("Full flow test", function () {
...
+        const filter = {
+            fromBlock: "0x0",
+            toBlock: "0x1",
+            address: testData.stablecoin.address
+        }
+
+        // needed to make web3j on backend work correctly with hardhat test network
+        for (let i = 0; i < 16; i++) {
+            await ethers.provider.send("eth_newFilter", [filter])
+        }
```
Comment (githinho) [inline test/full-test.ts:430]: Maybe extract this to function: `setupWeb3jFilter` or something similar

File: util/TestData.ts [hunk 3]
```diff
@@ -43,6 +45,7 @@ export class TestData {
 
     //////// CONTRACTS ////////
     stablecoin: Contract;
+    rewardStablecoin: Contract;
```
Comment (githinho) [inline util/TestData.ts:48]: I would rather call it `rewardCoin` to distinguish it more from stablecoin.

**Comment by domagojlatecki** [comment]
This pull request is temporarily on hold as there need to be some changes related to payouts as discussed with @mightymatth and @filipdujmusic. Another pull request will be made which will merge into this one to implement new changes.

**Comment by githinho** [comment]
Resolve conflicts

**Comment by domagojlatecki** [comment]
Conflicts are resolved and tests now pass - please review.

File: util/payout-service.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    message: string
+}
+
+export interface CreatePayoutResponse {
+    task_id: string
+}
+
+export type PayoutStatus = "PROOF_PENDING" | "PROOF_FAILED" | "PROOF_CREATED" | "PAYOUT_CREATED"
+
+export interface PayoutResponse {
```
Comment (githinho) [inline util/payout-service.ts:163]: @mightymatth here are payout responses so you can reuse them if needed.

**Review by githinho — APPROVED** [review]

**Pull request merged**